packages/contracts/src/contracts/current/utils/LibMem/LibMem.sol

```diff
@@ -0,0 +1,104 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.4.24;
+
+contract LibMem {
+
+    function getMemAddress(bytes memory input)
+        internal
+        pure
+        returns (uint256 address_)
+    {
+        assembly {
+            address_ := input
+        }
+        return address_;
+    }
+
+    /// @dev Writes a uint256 into a specific position in a byte array.
+    /// @param dest memory adress to copy bytes to
+    function memcpy(
+        uint256 dest,
+        uint256 source,
+        uint256 length
+    )
+        internal
+        pure
+    {
+        // Base cases
+        if(length == 0) return;
+        if(source == dest) return;
+
+        // Copy bytes from source to dest
+        assembly {
+            // Compute number of complete words to copy + remaining bytes
+            let lenFullWords := div(add(length, 0x1F), 0x20)
+            let remainder := mod(length, 0x20)
+            if gt(remainder, 0) {
+                lenFullWords := sub(lenFullWords, 1)
+            }
+
+            // Copy full words from source to dest
+            let offset := 0
+            let maxOffset := mul(0x20, lenFullWords)
+            for {offset := 0} lt(offset, maxOffset) {offset := add(offset, 0x20)} {
+                mstore(add(dest, offset),  mload(add(source, offset)))
```

Comment:
We can take one `add` out of the loop by having the loop run from `dest` to `dest + length`.